packages/webapp/src/components/Table/types.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import type { ReactElement, ReactNode } from 'react';
+import type { ColumnInstance } from 'react-table';
+import { ReactComponentLike } from 'prop-types';
+
+export enum TableType {
+  V1 = 'v1',
+  V2 = 'v2',
+}
+
+export enum CellType {
+  HOVER_PILL_OVERFLOW = 'hoverPillOverflow',
+  ICON_TEXT = 'iconText',
+  PLAIN = 'plain',
+  RIGHT_CHEVRON_LINK = 'rightChevronLink',
+}
+
+enum Alignment {
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+type CallbackFn = (props: any) => any;
+
+export type ReactTableProps = {
+  columns: ColumnInstance[];
+  data: Object[];
+  showPagination: boolean;
+  pageSizeOptions: number[];
+  defaultPageSize: number;
+  className: string;
+  getTdProps: Function;
+  sortByID: number;
+  minRows: number;
+  orderDesc: boolean;
+};
+
+type MuiTableColumn = {
+  id: string;
+  format: CallbackFn;
+  align: Alignment;
+  Footer: ReactElement;
+  columnProps: Object;
+  label: string;
+};
+
+export type MuiTableProps = {
+  alternatingRowColor: boolean;
+  columns: Required<MuiTableColumn[]>;
+  data: Object[];
+  showPagination: boolean;
+  pageSizeOptions: number[];
+  minRows: number;
+  dense: boolean;
+  FooterCell: ReactComponentLike;
+  onClickMore: CallbackFn;
+  itemsToAddPerLoadMoreClick: number;
+  onRowClick: CallbackFn;
+  shouldFixTableLayout: boolean;
+  defaultOrderBy: string;
+};
+
+export type PlainCellProps = {
+  text: string | null;
+};
+
+export type RightChevronLinkProps = {
+  path: string;
+};
+
+export type IconTextProps = {
+  text: string | null | undefined;
+  icon: FunctionReturnsComponent;
+  subtext: string | null | undefined;
+};
+
+// TODO: figure out how to return ReactNode/ Elemnet / Whatever - consider updating typescript
+export type FunctionReturnsComponent = (props: any) => any;
```

Comment:
Really struggled to get this right I feel like ReactNode should work but doesn't.

Reply:
This gives you a ts error right in the file?

```ts
export type FunctionReturnsComponent = (props: any) => ReactNode;
```

Or you mean it gives you an error when you use it in context? If the second, I think it's because an icon isn't a function that returns a ReactNode, it is just a ReactNode:

```ts
export type IconTextProps = {
  text?: string | null;
  icon: ReactNode;
  subtext?: string | null;
};
```

Reply:
I am having trouble keeping up with the flakiness of typescript haha. 

I though I was seeing other errors too in /V2 etc. Will go this route thanks!